bastion/src/path.rs

```diff
@@ -0,0 +1,459 @@
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // SYSTEM + smth
+
+    #[test]
+    fn append_sv_to_system() {
+        let sv_id = BastionId::new();
+        let path = BastionPath::root()
+            .append(BastionPathElement::Supervisor(sv_id.clone()))
+            .unwrap();
+        assert_eq!(path.iter().collect::<Vec<&BastionId>>(), vec![&sv_id]);
+    }
+
+    #[test]
+    fn append_children_to_system() {
+        let sv_id = BastionId::new();
+        let res = BastionPath::root().append(BastionPathElement::Children(sv_id));
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            "Children is not appendable to root"
+        );
+    }
+
+    #[test]
+    fn append_child_to_system() {
+        let sv_id = BastionId::new();
+        let res = BastionPath::root().append(BastionPathElement::Child(sv_id));
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            "Child is not appendable to root"
+        );
+    }
+
+    // Supervisor + smth
+
+    #[test]
+    fn append_sv_to_sv() {
+        let sv1_id = BastionId::new();
+        let sv2_id = BastionId::new();
+        let path = BastionPath::root()
+            .append(BastionPathElement::Supervisor(sv1_id.clone()))
+            .unwrap()
+            .append(BastionPathElement::Supervisor(sv2_id.clone()))
+            .unwrap();
+        assert_eq!(
+            path.iter().collect::<Vec<&BastionId>>(),
+            vec![&sv1_id, &sv2_id]
+        );
+    }
+
+    #[test]
+    fn append_children_to_sv() {
+        let sv_id = BastionId::new();
+        let children_id = BastionId::new();
+        let path = BastionPath::root()
+            .append(BastionPathElement::Supervisor(sv_id.clone()))
+            .unwrap()
+            .append(BastionPathElement::Children(children_id.clone()))
+            .unwrap();
+        assert_eq!(
+            path.iter().collect::<Vec<&BastionId>>(),
+            vec![&sv_id, &children_id]
+        );
+    }
+
+    #[test]
+    fn append_child_to_sv() {
+        let sv_id = BastionId::new();
+        let children_id = BastionId::new();
+        let res = BastionPath::root()
+            .append(BastionPathElement::Supervisor(sv_id))
+            .unwrap()
+            .append(BastionPathElement::Child(children_id));
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            "Child is not appendable to a supervisor"
+        );
+    }
+
+    // children + smth
+
+    #[test]
+    fn append_sv_to_children() {
+        let sv_id = BastionId::new();
+        let children_id = BastionId::new();
+        let res = BastionPath::root()
+            .append(BastionPathElement::Supervisor(sv_id))
+            .unwrap()
+            .append(BastionPathElement::Children(children_id))
+            .unwrap()
+            .append(BastionPathElement::Supervisor(BastionId::new()));
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            "Supervisor is not appendable to children"
```

Comment:
This is not true by our backed API definition. I think children should able to spawn other supervisors. @azastrael wdyt?

Reply:
They are able to spawn other supervisor. Path mirrors internal semantics.
AFAIK right now supervisor spawned by a child will have this child's supervisor as a parent. So path is `child_sv/spawned_by_child_sv`

Reply:
I think having a logical path is ok. That will help us a lot.

Reply:
You currently can't spawn supervisors directly from a child's context but you can do it using `Bastion` or a `SupervisorRef`.